cvxpy/tests/test_qp_solvers.py

```
@@ -445,7 +446,21 @@ def test_warm_start(self) -> None:
         result = prob.solve(solver="OSQP", warm_start=True)
         result2 = prob.solve(solver="OSQP", warm_start=False)
         self.assertAlmostEqual(result, result2)
-        pass
+
+    def test_gurobi_warmstart(self) -> None:
+        """Test Gurobi warm start with a user provided point.
+        """
+        if cp.GUROBI in INSTALLED_SOLVERS:
+            m = 20
+            n = 10
+            np.random.seed(1)
+            A = np.random.randn(m, n)
+            b = np.random.randn(m)
+
+            x = Variable(n, boolean=True)
+            prob = Problem(Minimize(sum_squares(A @ x - b)))
+            x.value = np.ones(n)
+            prob.solve(solver=cp.GUROBI, warm_start=True)
```

I have a similar feeling about this test and the last test; they should both be structured so that we're confident that options are getting passed correctly. I know it's a pain to write that kind of test, but I want to avoid situations like with our CBC interface (where it turns out we've been silently failing to set solver options for a long time).

In this case I would generate a problem where ``A`` is wide, ``b = A @ np.random.randint(-10, 10, n)``, and ``x.value = np.ones(n)``. If ``A`` is wide enough then I think this problem will be hard to solve, since most convex relaxations encountered in branch-and-bound will be able to achieve $Ax=b$, but probably not with integer solutions.